src/asm/lexer.c

```diff
@@ -167,24 +167,47 @@ static void yy_buffer_append(YY_BUFFER_STATE buf, uint32_t capacity, char c)
 	buf->pBufferStart[buf->nBufferSize] = 0;
 }
 
+static void yy_buffer_append_newlines(YY_BUFFER_STATE buf, size_t capacity)
+{
+	/* Add newline if file doesn't end with one */
+	if (buf->nBufferSize == 0
+	 || buf->pBufferStart[buf->nBufferSize - 1] != '\n')
+		yy_buffer_append(buf, capacity, '\n');
+
+	/* Add newline if \ will eat the last newline */
+	if (buf->nBufferSize >= 2) {
+		size_t pos = buf->nBufferSize - 2;
+
+		/* Skip spaces and tabs */
+		while (pos > 0 && (buf->pBufferStart[pos] == ' '
+				|| buf->pBufferStart[pos] == '\t'))
+			pos--;
+
+		if (buf->pBufferStart[pos] == '\\')
+			yy_buffer_append(buf, capacity, '\n');
+	}
+}
+
 YY_BUFFER_STATE yy_scan_bytes(char *mem, uint32_t size)
 {
 	YY_BUFFER_STATE pBuffer = malloc(sizeof(struct yy_buffer_state));
 
 	if (pBuffer == NULL)
 		fatalerror("%s: Out of memory!", __func__);
 
-	pBuffer->pBufferRealStart = malloc(size + 1 + SAFETYMARGIN);
+	size_t capacity = size + 3; /* space for 2 newlines and terminator */
+	pBuffer->pBufferRealStart = malloc(capacity + SAFETYMARGIN);
```

Comment:
Can you fix [the warning this causes](https://travis-ci.org/rednex/rgbds/jobs/583086153#L239)?

Reply:
I personally disagree with some of these coding style rules that `checkpatch` enforces... I don't mind changing it, though.

Reply:
Me neither, but it's there.